src/ptvsd/adapter/__main__.py

```diff
@@ -33,22 +33,27 @@ def main(args):
     log.describe_environment("ptvsd.adapter startup environment:")
 
     session = session.Session()
-    if args.debug_server is None:
+    if args.port is None:
         session.connect_to_ide()
     else:
         # If in debugServer mode, log everything to stderr.
         log.stderr_levels |= set(log.LEVELS)
-        with session.accept_connection_from_ide(("localhost", args.debug_server)):
-            pass
+
+        if args.connect_to_port is not None:
+            session.connect_to_server(("127.0.0.1", args.connect_to_port))
+        with session.accept_connection_from_ide((args.host, args.port)) as (_, port):
+            try:
+                session.server.set_debugger_property(adapterPort=port)
+            except AttributeError:
+                pass
     session.wait_for_completion()
 
 
 def _parse_argv(argv):
     parser = argparse.ArgumentParser()
 
     parser.add_argument(
-        "-d",
-        "--debug-server",
+        "--port",
```

Comment:
I think at this point you need to remove `nargs` and `const` below. What they did is allow to specify `--debugServer` without any value, and then `const` would be the implied value. But this doesn't make any sense for `--port` and `--host`.